src/components/Integrations/BigQuery/ControlPanel/test/InputSpanForm.test.js

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import { render, fireEvent, screen, waitFor } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import IntlProvider from '../../../../../i18n/DopplerIntlProvider.double-with-ids-as-values';
+import { InputSpanForm } from '../InputSpanForm';
+import { debug } from 'console';
+
+describe('test for validate input span component ', () => {
+  test('Validate loading message disappear', async () => {
+    const { container } = render(
+      <IntlProvider>
+        <InputSpanForm />
+      </IntlProvider>,
+    );
+    expect(container.getElementsByClassName('loading-box').length).toBe(1);
+    await waitFor(() => {
+      setTimeout(function () {
+        expect(container.getElementsByClassName('loading-box').length).toBe(0);
+      }, 3000);
+    });
+  });
+
+  test('Validate empty input', () => {
+    const { getByTestId } = render(
+      <IntlProvider>
+        <InputSpanForm />
+      </IntlProvider>,
+    );
+
+    const emptyInput = screen.getByTestId('email-input');
+    expect(emptyInput.value).toBe('');
+  });
+
+  test('Validate button disabled when input is empty', () => {
+    const { getByTestId } = render(
+      <IntlProvider>
+        <InputSpanForm />
+      </IntlProvider>,
+    );
+
+    const buttonAdd = getByTestId('btn-add');
+    expect(getByTestId('btn-add')).toBeDisabled();
+  });
+
+  test('Validate button enabled when input is not empty', () => {
+    const { getByTestId } = render(
+      <IntlProvider>
+        <InputSpanForm />
+      </IntlProvider>,
+    );
+    const emptyInput = screen.getByTestId('email-input');
+    fireEvent.change(emptyInput, { target: { value: 'input not empty' } });
+    const buttonAdd = getByTestId('btn-add');
+    expect(getByTestId('btn-add')).toBeEnabled();
+    screen.debug();
```

Comment:
**`debug`** should be removed, because they print to **`console`**

Reply:
solved in next push

Reply:
Hecho, resuelvo la conversación...